.github/workflows/benchmark.yml

```diff
@@ -103,33 +116,56 @@ jobs:
     permissions:
       pull-requests: write
       contents: write
+    env:
+      BENCHER_PROJECT: tailcall
+      BENCHER_ADAPTER: rust_criterion
+      BENCHER_TESTBED: benchmarking-runner
+      PR_BENCHMARK_RESULTS: benches/change.txt
+      BASE_BENCHMARK_RESULTS: benches/base.txt
+      UPPER_BOUNDARY: 0.10
     steps:
       - name: Check out code
         uses: actions/checkout@v4
         with:
           ref: ${{ github.event.pull_request.head.sha }}
 
-      - name: Install Rust
-        uses: actions-rust-lang/setup-rust-toolchain@v1
-
-      - name: Run Criterion Benchmarks
-        run: |
-          cargo install cargo-criterion rust-script
-          cargo criterion --message-format=json > benches/benchmarks.json
-          ./scripts/json_to_md.rs benches/benchmarks.json > benches/change_benchmarks.md
-
-      - name: Print Criterion Benchmarks
-        run: cat benches/change_benchmarks.md
-
       - name: Restore file
         uses: actions/cache@v4
         with:
-          path: benches/main_benchmarks.json
+          path: benches/base.txt
           key: criterion_benchmarks_${{ github.event.pull_request.base.sha }}
           fail-on-cache-miss: true
 
-      - name: Print Benchmark Comparision
-        run: ./scripts/criterion_compare.rs benches/main_benchmarks.json benches/benchmarks.json table
+      - name: Run Benchmarks
+        run: |
+          cargo bench > benches/change.txt
+
+      - name: Install Bencher CLI
+        uses: bencherdev/bencher@main
 
-      - name: Check Degradation
-        run: ./scripts/criterion_compare.rs  benches/main_benchmarks.json benches/benchmarks.json check
+      - name: Track base benchmarks
+        run: |
+          bencher run \
+          --if-branch '${{ github.head_ref }}_commit_${{ github.event.pull_request.head.sha }}' \
```

Comment:
having same branch name for each threshold returns an error so i am changing branch name with each commit